lib/bindings/HTTPBinding.js

```diff
@@ -0,0 +1,283 @@
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with iotagent-json.
+ * If not, seehttp://www.gnu.org/licenses/.
+ *
+ * For those usages not covered by the GNU Affero General Public License
+ * please contact with::[contacto@tid.es]
+ */
+var iotAgentLib = require('iotagent-node-lib'),
+    intoTrans = iotAgentLib.intoTrans,
+    _ = require('underscore'),
+    logger = require('logops'),
+    async = require('async'),
+    commandHandler = require('../commandHandler'),
+    request = require('request'),
+    errors = require('../errors'),
+    express = require('express'),
+    http = require('http'),
+    commonBindings = require('../commonBindings'),
+    bodyParser = require('body-parser'),
+    constants = require('../constants'),
+    apply = async.apply,
+    context = {
+        op: 'IoTAgentJSON.HTTPBinding'
+    },
+    config = require('../configService'),
+    httpBindingServer;
+
+function checkMandatoryParams(queryPayload) {
+    return function(req, res, next) {
+        var notFoundParams = [],
+            error;
+
+        req.apiKey = req.query.k;
+        req.deviceId = req.query.i;
+
+        if (!req.apiKey) {
+            notFoundParams.push('API Key');
+        }
+
+        if (!req.deviceId) {
+            notFoundParams.push('Device Id');
+        }
+
+        if (queryPayload && !req.query.d) {
+            notFoundParams.push('Payload');
+        }
+
+        if (req.method === 'POST' && !req.is('json')) {
+            error = new errors.UnsupportedType('application/json');
+        }
+
+        if (notFoundParams.length !== 0) {
+            next(new errors.MandatoryParamsNotFound(notFoundParams));
+        } else {
+            next(error);
+        }
+    };
+}
+
+function parseData(req, res, next) {
+    var data,
+        error,
+        payload;
+
+    if (req.body) {
+        data = req.body;
+    } else {
+        payload = req.query.d;
+
+        logger.debug(context, 'Parsing payload [%s]', payload);
+
+        try {
+            data = JSON.parse(payload);
+        } catch (e) {
+            error = e;
+        }
+    }
+
+    if (error) {
+        next(error);
+    } else {
+        req.jsonPayload = data;
+
+        logger.debug(context, 'Parsed data: [%j]', data);
+        next();
+    }
+}
+
+function executeCommand(apiKey, device, serializedPayload, callback) {
+    var options = {
+        url: device.endpoint,
+        method: 'POST',
+        body: serializedPayload,
+        headers: {
+            'fiware-service': device.service,
+            'fiware-servicepath': device.subservice
+        }
+    };
+
+    request(options, function(error, response, body) {
+        if (error || response.statusCode !== 200) {
+            callback(new errors.HTTPCommandResponseError(response.statusCode, error));
+        } else {
+            process.nextTick(commandHandler.updateCommand.bind(null, apiKey, device, body, callback));
+        }
+
+        callback();
```

Comment:
This `callback()` invocation may lead to unexpected behaviours as a consequence of the `callback()` being invoked twice and not with the expected arguments, doesn't it?